timemachine/cpp/src/kernels/k_harmonic_angle.cuh

```diff
@@ -45,16 +48,19 @@ void __global__ k_harmonic_angle(
         njk += vjk * vjk;
     }
 
-    nji = sqrt(nji);
-    njk = sqrt(njk);
+    rji[3] = eps;
+    rjk[3] = eps;
+
+    nji = sqrt(nji + eps * eps);
+    njk = sqrt(njk + eps * eps);
 
     RealType hi = 0;
     RealType lo = 0;
 
     // second pass, compute the hi/lo values
-    for (int d = 0; d < 3; d++) {
-        RealType vji = rji[d]; // coords[i_idx * 3 + d] - coords[j_idx * 3 + d];
-        RealType vjk = rjk[d]; // coords[k_idx * 3 + d] - coords[j_idx * 3 + d];
+    for (int d = 0; d < 4; d++) {
+        RealType vji = rji[d]; // coords[i_idx * D + d] - coords[j_idx * D + d];
+        RealType vjk = rjk[d]; // coords[k_idx * D + d] - coords[j_idx * D + d];
```

Comment:
[inconsequential] doc error

Reply:
addressed in https://github.com/proteneer/timemachine/pull/1520/commits/3a79f0791d9b40d692023a8ae859c148bf2cbd15

(replaced D with the hard coded value of 4)

Reply:
brainfart on my end!

Reply:
non-sensical comment removed. Thanks for catching this.